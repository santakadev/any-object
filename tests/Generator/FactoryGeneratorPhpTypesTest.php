<?php

declare(strict_types=1);

namespace Santakadev\AnyObject\Tests\Generator;

use ApprovalTests\Approvals;
use DateTime;
use DateTimeImmutable;
use Santakadev\AnyObject\AnyObject;
use Santakadev\AnyObject\Tests\Generator\Generated\AnyDateTimeImmutableObject;
use Santakadev\AnyObject\Tests\Generator\Generated\AnyDateTimeObject;
use Santakadev\AnyObject\Tests\TestData\PhpTypes\DateTimeImmutableObject;
use Santakadev\AnyObject\Tests\TestData\PhpTypes\DateTimeObject;

class FactoryGeneratorPhpTypesTest extends FactoryGeneratorTestCase
{
    /** @dataProvider anyProvider */
    public function test_date_time(AnyObject $any): void
    {
        $this->generateFactoryFor(DateTimeObject::class);

        $text = $this->readGeneratedAnyFileFor(DateTimeObject::class);
        Approvals::verifyString($text);
        $object = AnyDateTimeObject::build();
        $this->assertInstanceOf(DateTime::class, $object->value);
    }

    /** @dataProvider anyProvider */
    public function test_date_time_immutable(AnyObject $any): void
    {
        $this->generateFactoryFor(DateTimeImmutableObject::class);

        $text = $this->readGeneratedAnyFileFor(DateTimeImmutableObject::class);
        Approvals::verifyString($text);
        $object = AnyDateTimeImmutableObject::build();
        $this->assertInstanceOf(DateTimeImmutable::class, $object->value);
    }
}
