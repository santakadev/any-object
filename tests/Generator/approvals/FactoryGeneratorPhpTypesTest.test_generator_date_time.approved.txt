<?php

namespace Santakadev\AnyObject\Tests\Generator\Generated;

use Faker\Factory;
use Santakadev\AnyObject\Tests\TestData\PhpTypes\DateTimeObject;
use DateTime;

final class AnyDateTimeObject
{
    public static function with(DateTime|None $value = new None()) : DateTimeObject
    {
        if ($value instanceof None) {
            $faker = Factory::create();
            $value = new DateTime();
        }
        return new DateTimeObject($value);
    }
    public static function build() : DateTimeObject
    {
        return self::with();
    }
}
